osbuild/util/ostree.py

```diff
@@ -215,16 +215,30 @@ def parse_input_commits(commits):
     return commits["path"], data["refs"]
 
 
-def deployment_path(root: PathLike, osname: str = "", ref: str = "", serial: int = None):
-    """Return the path to a deployment given the parameters"""
+def parse_deployment_option(root, deployment):
+    """Parse the deployment option and return the osname, ref, and serial"""
 
-    if osname == "" and ref == "" and serial is None:
+    default_deployment = deployment.get("default", False)
+    if default_deployment:
         filenames = glob.glob(os.path.join(root, 'ostree/deploy/*/deploy/*.0'), recursive=True)
         if len(filenames) < 1:
             raise ValueError("Could not find deployment")
         if len(filenames) > 1:
             raise ValueError("More than one deployment found")
-        return filenames[0]
+        
+        path = filenames[0].split("/")
+        osname = path[-3]
+        ref, serial = path[-1].split(".")
```

Comment:
I think there may be a better way to pull out the pieces here using a regex.. at least it may be a little easier to understand when reading the code:

```
>>> filenames = ["/ostree/deploy/fedora-coreos/deploy/aa621168d0a62f56ed8591e35c6cebd29bd57da7f77f3dd720477eb586c0c4ac.0"]
>>> r = re.compile("/ostree/deploy/(.*)/deploy/(.*)\\.([0-9])")
>>> f = r.match(filenames[0])
>>> print(f"0: {f.group(0)}\n1: {f.group(1)}\n2:{f.group(2)}\n3:{f.group(3)}")
0: /ostree/deploy/fedora-coreos/deploy/aa621168d0a62f56ed8591e35c6cebd29bd57da7f77f3dd720477eb586c0c4ac.0
1: fedora-coreos
2:aa621168d0a62f56ed8591e35c6cebd29bd57da7f77f3dd720477eb586c0c4ac
3:0
```